tests/test_check_parallel.py

```diff
@@ -174,8 +177,22 @@ def teardown_class(self):
 
     def test_worker_initialize(self) -> None:
         linter = PyLinter(reporter=Reporter())
-        worker_initialize(linter=linter)
-        assert pylint.lint.parallel._worker_linter == linter
+        worker_initialize(linter=dill.dumps(linter))
+        assert isinstance(pylint.lint.parallel._worker_linter, type(linter))
+
+    def test_worker_initialize_pickling(self) -> None:
+        """Test that we can pickle objects that standard pickling in multiprocessing can't.
+
+        See:
+        https://stackoverflow.com/questions/8804830/python-multiprocessing-picklingerror-cant-pickle-type-function
+        https://github.com/PyCQA/pylint/pull/5584
+        """
+        linter = PyLinter(reporter=Reporter())
+        linter.attribute = argparse.ArgumentParser()  # type: ignore[attr-defined]
+        pool = multiprocessing.Pool(  # pylint: disable=consider-using-with
```

Comment:
You don't want to use `with` ? :)

Reply:
This was just directly copied from `parallel.py`. We can change, but I think by keeping it similar it shows better what we are testing here.

Reply:
Actually I'll do a follow-up PR where I'll add a `with` to `parallel.py` and this test (if this gets merged).